ui/src/components/TabbedLayout.tsx

```diff
@@ -1,14 +1,46 @@
-import React from 'react';
 import {PageSection, PageSectionVariants, Tab, Tabs, TabTitleText,} from '@patternfly/react-core';
 import {DepCompoundTable} from "./DepCompoundTable";
 import {getSourceName, getSources, Report} from "../api/report";
+import {AnalyticsBrowser, AnalyticsBrowserSettings} from '@segment/analytics-next'
+import React, {useEffect, useRef} from 'react';
+import {useAppContext} from '../App';
 
 export const TabbedLayout = ({report}: { report: Report }) => {
+  const appContext = useAppContext();
   const sources = getSources(report);
-
   const [activeTabKey, setActiveTabKey] = React.useState<string | number>(getSourceName(sources[0]));
   const [isTabsLightScheme] = React.useState<boolean>(true);
 
+  const analytics = appContext.writeKey && appContext.writeKey.trim() !== '' ?
+      AnalyticsBrowser.load({ writeKey: appContext.writeKey } as AnalyticsBrowserSettings) : null;
+  const previousUserId = useRef<string | null>(null);
+  const previousActiveTabKey = useRef<string | number>('');
+
+    useEffect(() => {
+      if (!analytics) return;
+
+      if (appContext.userId == null) {
+        if (appContext.anonymousId != null) {
+          analytics.setAnonymousId(appContext.anonymousId);
+        }
+      } else {
+        if (appContext.userId !== previousUserId.current) {
+          analytics.identify(appContext.userId);
```

Comment:
The `identify` event should've been sent by the backend. You shouldn't need this. This event will override the existing user-related data.

Reply:
I'll remove it.

Reply:
Identify() does not override